browser-test/src/address.test.ts

```diff
@@ -0,0 +1,248 @@
+import {AdminPrograms, AdminQuestions, ApplicantQuestions, loginAsAdmin, loginAsGuest, logout, resetSession, selectApplicantLanguage, startSession} from './support'
+
+describe('address applicant flow', () => {
+  let pageObject;
+
+  beforeAll(async () => {
+    const {page} = await startSession()
+    pageObject = page;
+  });
+
+  afterEach(async () => {
+    await resetSession(pageObject);
+  });
+
+  describe('single address question', () => {
+    let applicantQuestions;
+    const programName = 'test program for single address';
+
+    beforeAll(async () => {
+      await loginAsAdmin(pageObject);
+      const adminQuestions = new AdminQuestions(pageObject);
+      const adminPrograms = new AdminPrograms(pageObject);
+      applicantQuestions = new ApplicantQuestions(pageObject);
+
+      await adminQuestions.addAddressQuestion({questionName: 'address-test-q'});
+      await adminPrograms.addAndPublishProgramWithQuestions(['address-test-q'], programName);
+
+      await logout(pageObject);
+    });
+
+    it('does not show errors initially', async () => {
+      await loginAsGuest(pageObject);
+      await selectApplicantLanguage(pageObject, 'English');
+
+      await applicantQuestions.applyProgram(programName);
+      await applicantQuestions.answerAddressQuestion('1234 St', 'Unit B', 'Sim', 'Ames', '54321');
+      let error = await pageObject.$('.cf-address-street-1-error');
+      expect(await error.isHidden()).toEqual(true);
+      error = await pageObject.$('.cf-address-city-error');
+      expect(await error.isHidden()).toEqual(true);
+      error = await pageObject.$('.cf-address-state-error');
+      expect(await error.isHidden()).toEqual(true);
+      error = await pageObject.$('.cf-address-zip-error');
+      expect(await error.isHidden()).toEqual(true);
+    });
+
+    it('with valid address does submit', async () => {
+      await loginAsGuest(pageObject);
+      await selectApplicantLanguage(pageObject, 'English');
+
+      await applicantQuestions.applyProgram(programName);
+      await applicantQuestions.answerAddressQuestion('1234 St', 'Unit B', 'Sim', 'Ames', '54321');
+      await applicantQuestions.clickNext();
+
+      await applicantQuestions.submitFromReviewPage(programName);
+    });
+
+    it('with invalid address does not submit', async () => {
```

Comment:
Maybe change to "empty address" and add another test for an invalid address - i.e. incorrect zip code.